x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/aggregation/AggregationToJsonProcessor.java

```diff
@@ -300,6 +303,14 @@ private boolean addMetricIfFinite(String key, double value) {
         return false;
     }
 
+    private boolean processGeoCentroid(GeoCentroid agg) {
+        if (agg.count() > 0) {
```

Comment:
I did not think it was necessary for the `GeoCentroid#count` to change the overall `doc_count` recorded. It seemed logical to me that we still rely on the bucket to keep track of the docs. 

The only downside that I do not know how to really address is if there are certain docs within the bucket that do not have points and thus the `GeoCentroid#count` would be less than the document count. Since I am not sure how `lat_long` is handled on the back end, I opted for the simplest solution.